### ISSUE TITLE
[Suggestion] Proto Life Support

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#1817]
I don't know what is the plan for life support. A while back there would be no air pressurization, but later I heard something else. So here is an early solution for life support if not a permanent solution.

The life support will be just like the gravity block. Something you place and that can be configured to extend to an area. Another option is the entire ship so for larger ships you wouldn't have to put down dozens of blocks in say a 100m ship. Something I would also like for the gravity block as well. Bounding boxes and all of that. Plus it will stop players from making bubbles of gravity or life support outside of the ship. And to be honest the UI for configuring the gravity field isn't very user friendly. Got to make one change, look around, make another, look around over and over again. 

![](https://cdn.discordapp.com/attachments/406664114249793536/702777739392057466/unknown.png)

Anything inside the range can breathe, everyone outside will slowly take damage then die. In real life you lose consciousness in 15 seconds without a suit in space. The life support block will use power relative to its size just like the gravity generator.

![](https://cdn.discordapp.com/attachments/271723356795961345/702779602573066310/unknown.png)

Finally there is considering the future space suits and how they would regain their oxygen for EVA. For that I would suggest going for the Subnautica model. In Subnautica when you go under water, you can stay down there for a certain amount of time before you start to drown. Coming above water will fill up your oxygen. In SEVO it would be the same thing. Go inside the life support area and refill your oxygen. Like Subnatica, you can upgrade your oxygen tanks for more time in space.

In my option pressurization besides being very complicated in a game like this would just add unnecessary troubles. I don't want to worry about breathing when my ship's hull inevitably gets pierced and all the oxygen in the ship gets sucked out. Works in the Expanse, not as much here.

**Comment by tsunamayo** [comment]
okay before I do a bigger answer, how gravity is working is Starmade / SE (if you know).
I can improve the current visualization / experience. I dont want to make a global setting, it is lazy and no fun possibility... plus what is inside? You will still have bubble around your ship with a global setting.
If you know what is inside (an algo that says inside / outside) then congrats you just got yourself an air-pressure system ;) ...

**Comment by Xerma** [comment]
Space Engineers as far as I know only uses gravity generators and they only affect an area sort of like how we currently have but their setting can also be tweaked. Most people end up modding this because it's just clunky to work around for proper gravity on ships. Starmade just uses a single block with no power consumption for global gravity on the entire ship. A good work around could be a mix of what we have with global gravity. What could work is have draggable gravity systems like how reactors or shields work, they require power, give off heat, etc but as you get a bigger system the area gets larger and requires more power, heat, etc. This could have a max amount of lets say 6 x 6 x 6 (just an example) and maybe that gravity generators range is 32 meters from each face of the system.

Something like that would really fit the feel of the game in my opinion and is a compromise between a global gravity setting vs numerous gravity generators which can get very tedious very fast. Especially for ships larger than 100m - also depending on their total volume.

**Comment by ZachZent** [comment]
Star Made: Press R on a gravity block and you will have gravity around you.
Space Engineers: For only the player, a grav block is placed and a shape, square or circle is chosen. I don't know much more about it.

![](https://cdn.discordapp.com/attachments/271723356795961345/702789895239761950/unknown.png)

I think you are confusing lazy with tedious. Placing a bunch of grav gens and having to manually enter each one is not exactly fun. Especially for a larger ship where you have to make sure you fill up every space. It could take a dozen to fully grav up a decently sized ship given that we can only do a max field of 16x16x16. I agree that being able to have areas that are a different gravity is fun, but manually inputting each value over and over again is not fun or creative. Like life support, I would rather not have to think about gravity after I place it, minus the times I decide to turn it off and instead focus on the actual design of the ship.

And when I was meaning for pressurization is actual pressure with air flying out the holes of the ship. Life support that encompasses the entire ship's "inside" is perfectly fine.

**Comment by tsunamayo** [comment]
No I was saying lazy game-design wise, on my side. I get it is tedius I want to improve it... Lets open a new thread on gravity I guess. Thanks for the explanation!

**Comment by tsunamayo** [comment]
So I would prefer a true air-pressure system. This one of the reason I also put blast door. So much fun possibilities for crew and boarding, pushing crew outside ect. 
You have a helmet so you can survive in space. I wanted it to have impact on crew management: if a room is not pressured, crew cant boost their system or control their turret. They can only repair.
Plus I would like fire and incendiary shell as anti-crew, so you could do like in FTL: open the door and kill the fire ;)
This is a typically Fun vs other constrain choice.
Also I will need air computation if I ever want boat and submarines ;)

**Comment by Sullos** [comment]
I can certainly see the benefit and interest of a true air-pressure system. The difficult point is having to choose when you might need to tell people a limitation of their building. Like to make sure all rooms have walls made of hull blocks, with a blast door as the only entry. It all depends on what you're capable of doing. And also what's possible without putting too much strain on the game with large amounts of room air-pressure computations.
We can all offer suggestions on how it should operate, but it all depends on what you think is best from a coding perspective.

**Comment by ZachZent** [comment]
Personally I don't like the idea of needing to vent to go into combat. It works for the Expanse due to the physics and the literal hours it takes for a ship to come into range (minus stealth ships) so you have time to prepare compared to here where ships can jump in right next to you. HOWEVER after discussing this in the Discord, I see the benefits for what you are saying Tsuna.

This still works with my idea for a block like life support system similar to the gravity gen. I do know you are less keen on the idea of a total ship gravity field, but for life support it is kind of necessary. Can't really change directions for oxygen.

**Comment by FelixGold** [comment]
Could have a simple brick vent that you add to rooms, with the size of it determining how fast it can pressurize/depressurize a room, so you would want a large vent for airlocks, but some smaller ones for crew quarters and the like

**Comment by tsunamayo** [comment]
Yes actually we would need a block (maybe smaller) to provide air in each room. There would not be any direction, it would just provide air to nearby block that are not connected to void.
But again not saying this will be in the game 100% sure, this is what I want to do, I need to think more about the technology behind.

**Comment by Battlepixel** [comment]
" ...You have a helmet so you can survive in space. I wanted it to have impact on crew management: if a room is not pressured, crew cant boost their system or control their turret. They can only repair. .."
In my opinion that air only affect crew management ist a great idea. 
Please consider the following!!!:
It would be infinitely terrible not to be able to build small fighters because they need blast doors or have to use standart-cockpit-blocks. On small fighters you only need a pilot and maybe one gunner but they are mostly sitting. Would be best they dont need air to do their jobs otherwise all the infinite possibilities (we have now! )to build your own small cockpits and fighters would no longer be so infinite. For all bigger ships like corvettes frigates etc. its fine. By the way the possibility that my NPC crew repairs at least bigger ships sounds fantastic. For me it isnt fun to spend hours between battles to search an repair damage. If it did, I would have more pleasure repairing a bike in real life or something. What I cannot do in real life is to fight space battles with self-designed spaceships. That is what turns me and many of my friends off to SE. Micromanagement is over the top, there is no real AI to engage, and it takes hours to experience a little bit of action.

**Comment by FelixGold** [comment]
Thats a simple solution, the fighter pilots just carry air in their own inventory, rather than having the ship provide it. Alternatively sitting in a chair could provide oxygen through hookups.

I rather love the micromanagement part of SE and Empyrion, thats what makes survival actually fun and why none of my friends or I play in creative except to experiment with builds.

**Comment by Battlepixel** [comment]
How do you want to make sure that your npcs fly back into the hangar and then belong to the airlock to take a deep breath.^^ But if the chair provide air, it will do.

**Comment by FelixGold** [comment]
Well, NPC fighters would ideally keep their tanks/supplies topped off in their craft and suits, could be as simple as having a block/brick thats like a suit recharger, NPCs could be programmed with a priority to refill their suits tanks if their fuel/oxygen gets below a certain point, and to path to the nearest friendly recharge station to use it, even if it means returning to their mothership to do so same as low fuel/ammo for their fighter.

**Comment by tsunamayo** [comment]
@ZachZent btw could you send me you updated shuttle? Its design grew on me, I used to spawn it and destroy it when I was testing the targetting system. Was it posted on the pathfinding build? Thanks

**Comment by ZachZent** [comment]
Do you mean this shuttle?
http://skywa.re/blueprint/pahan-shuttle-combat-shuttle-variant

I did really like the design and am scaling it up a bit for a new ship. This picture is an older version, I can get a more updated picture when I get home from work. I also am going to submit this ship type for the police ship competition. Well after I paint replace the orange with blue
![](https://cdn.discordapp.com/attachments/702281585138991175/703086631401291856/unknown.png)

**Comment by tsunamayo** [comment]
Yeah exactly, design grew on me. I was also thinking about having this as a police ship ;)  I think you could make some interesting color scheme on that.

**Comment by ZachZent** [comment]
The smaller one has a full interior. I would be happy to build more of this style of ship

**Comment by tsunamayo** [comment]
@ZachZent does this is too sensitive because you cant make smaller side thrusters ? (because of the locked scaling?)

**Comment by ZachZent** [comment]
Bingo. They were supposed to fit in the slot and only be a few bricks tall, give or take. But since you can't Z scale the thrusters anymore, had to make do

**Comment by ZachZent** [comment]
![](https://cdn.discordapp.com/attachments/406664114249793536/703486611068813323/unknown.png)

Here is a more recent version of the larger ship. Hull is done, now detailing and paint

**Comment by tsunamayo** [comment]
I love the look, but I would make a larger version a bit more different, they are too similar looking I would say, I dont know how you could vary it though... I think factions ship ought to be a bit more than scaled up version of various ship. Easier said than done ;)
